### PR TITLE
Exposed `find_next_valid_focus` and `find_prev_valid_focus`.

### LINKED ISSUE #29357: Expose Control closest focus neighbour selection to GDScript
**Godot version:** 3.1.1 stable

**Issue description:**
This issue is relevant for making keyboard/controller-friendly UI navigation. Godot by default can determine the closest focus neighbours when the user tries to navigate the interface using default input actions. If the `focus_neighbour_*` properties are not explicitly set, quoting, "Godot will give focus to the closest Control to the [direction] of this one."

The problem comes when it is desired to call for this "best guess" on demand through script. To my understanding, there is currently no way to to obtain the closest neighbouring control node to give focus to. This functionality isn't exposed in the GDScript API, which makes it difficult to implement certain types of interfaces or for ease of use when navigating them with keyboard/controller.

I am asking for a function possibly called `get_closest_focus_neighbour` which would need a direction parameter and would return the closest node to give focus to, using the same algorithm Godot does when searching for them. This could also be extended to `focus_previous/next` best guess algorithms.

### PR DESCRIPTION
**Pull request by linkpy**
Fixes #29357.

### CODE REVIEW

**Comment by KoBeWi** [comment]
If you expose new methods you need to add documentation for them. You can generate missing docs by using `godot --doctool .` and filling the created description fields in `Control.xml` (other XML files might get modified too; you can discard these changes).

**Comment by linkpy** [comment]
The documentation is added, but there is an issue : 

```gd
var c: Control = # ...
c.find_ # < the two functions aren't shown in the autocomplete
```

I ran `bin/godot.linuxbsd.tools.64 --doctool .` and it also changed other .xml files at the same time.
I dont know if I forgot to do something.

**Comment by KoBeWi** [comment]
> the two functions aren't shown in the autocomplete

This might be some autocompletion bug, you exposed the methods correctly.

File: doc/classes/Control.xml [hunk 1]
```diff
@@ -715,6 +715,20 @@
 				Give up the focus. No other control will be able to receive keyboard input.
 			</description>
 		</method>
+        <method name="find_next_valid_focus" qualifiers="const">
```
Comment (pycbouh) [inline doc/classes/Control.xml:718]: You have spaces in this snippet and the file is formatted with tabs. Also, please put the new methods into the doc in the alphabetical order.

**Review by YuriSizov — CHANGES_REQUESTED** [review]

File: doc/classes/Control.xml (hunk 1 above)
Comment (linkpy) [reply doc/classes/Control.xml]: Documentation fixed.

**Review by YuriSizov — APPROVED** [review]

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you squash the three commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**